src/local_scheduler/local_scheduler.cc

```diff
@@ -138,7 +138,12 @@ void kill_worker(LocalSchedulerState *state,
     /* Update the task table to reflect that the task failed to complete. */
     if (state->db != NULL) {
       Task_set_state(worker->task_in_progress, TASK_STATUS_LOST);
+#if !RAY_USE_NEW_GCS
       task_table_update(state->db, worker->task_in_progress, NULL, NULL, NULL);
+#else
+      RAY_CHECK_OK(TaskTableAdd(&state->gcs_client, worker->task_in_progress));
+      Task_free(worker->task_in_progress);
```

Comment:
TaskTableAdd is async; is immediately calling Task_free safe?

Reply:
Yes, since it copies out all the relevant fields from the Task datastructure. The idea is that the Task datastructure will go away at some point and be replaced by TaskTableDataT.